src/natcap/invest/datastack.py

```diff
@@ -341,8 +355,12 @@ def _recurse(args_param, handler, nested_key=None):
                                  possible_path, filepath)
                     return filepath
                 except KeyError:
+                    # turn the nested key into a nice name for a folder
+                    # e.g. args['category']['data_path'] --> category_data_path
+                    folder_prefix = nested_key[6 : -2].replace('\'][\'', '_')
```

Comment:
I'm using the nested_key here because, if there were to be a nested params dictionary, it would uniquely identify each param, making for a helpful folder name.
However, it's not ideal if there is a list... see above

Reply:
See above my comments on lists. I think this replacement looks good!